api/src/Security/AccessTokenHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Security;
+
+use App\JWT\Validation\Constraint\HasClaim;
+use Lcobucci\Clock\SystemClock;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\UnencryptedToken;
+use Lcobucci\JWT\Validation\Constraint\LooseValidAt;
+use Lcobucci\JWT\Validation\Constraint\SignedWith;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Http\AccessToken\AccessTokenHandlerInterface;
+
+/**
+ * Parse and validate token manually.
+ * Decorates KeycloakAccessTokenHandler on dev environment for local usage.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ *
+ * @final
+ */
+class AccessTokenHandler implements AccessTokenHandlerInterface
+{
+    public function __construct(private readonly AccessTokenHandlerInterface $decorated, private readonly Configuration $configuration)
+    {
+    }
+
+    public function getUserIdentifierFrom(string $accessToken): string
+    {
+        try {
+            try {
+                return $this->decorated->getUserIdentifierFrom($accessToken);
+            } catch (\Throwable) {
```

Comment:
We should check the error. If Keycloak says that this token isn't valid, we should not try to verify it locally. For instance, the token can be not expired but invalidated server-side because the user changed is password.

Reply:
The `AccessTokenHandler` is for local development usage only (commented on line 18 on this class).

Running the project locally using Docker Compose, the user generates its token through `https://localhost:8080` domain to access Keycloak. This domain is stored in the token as `iss` token (as far as I remember).

When the API sends this token to Keycloak for validation, this must be done using the same domain the user called Keycloak: `https://localhost:8080`. Because of Docker network, the API cannot call Keycloak using this domain, it must call it through `keycloak` domain. Doing to invalidates the token and breaks the authentication/authorization workflow.

To prevent it on deployment, the k8s chart contains 2 domains totally separated for the API vs Keycloak. This problem is only for local development using Docker.

Another solution would be to ask developers to declare 2 different local domains and handle a alternative network configuration, but this would make the demo project installation definitively more complex.

Reply:
> Another solution would be to ask developers to declare 2 different local domains and handle a alternative network configuration, but this would make the demo project installation definitively more complex.

I think it's worth it

Reply:
We should find a way to run Keycloak behind Caddy as a proxy, and make it accessible through the same domain from the user and the API container